### PR TITLE
Updating workflows

### PR DESCRIPTION
**Pull request by PaurushGarg**
*Issue #, if available:*

*Description of changes:*
Fixing this PR: https://github.com/aws-observability/aws-otel-java-instrumentation/pull/627
1. Separate out contract tests for main and nightly build.
2. Remove patch build for nightlybuild workflow as this workflow should need patching.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/nightly-upstream-snapshot-build.yml [hunk 1]
```diff
@@ -45,22 +45,6 @@ jobs:
...
-      # cache local patch outputs
-      - name: Cache local Maven repository
-        id: cache-local-maven-repo
-        uses: actions/cache@v3
-        with:
-          path: |
-            ~/.m2/repository/io/opentelemetry/
-          key: ${{ runner.os }}-maven-local-${{ hashFiles('.github/patches/opentelemetry-java*.patch') }}
-
-      - name: Publish patched dependencies to maven local
```
Comment (vasireddy99) [inline .github/workflows/nightly-upstream-snapshot-build.yml:57]: the reason for removing is that patching is not required in the nightly build workflows as it testing against upstream snapshot?

File: .github/workflows/nightly-upstream-snapshot-build.yml [hunk 2]
```diff
@@ -115,24 +99,38 @@ jobs:
       test_ref: 'terraform'
       caller-workflow-name: 'nightly-upstream-snapshot-build'
 
-  # E2E tests where SampleApp has Java Agent
```
Comment (PaurushGarg) [inline .github/workflows/nightly-upstream-snapshot-build.yml:118]: Removed this Test as it builds the sample app with the released upstream dependencies and not with the snapshot one. I think providing this test here, requires much more work to ensure sample apps are build with java agent that have upstream snapshot versions.

Comment (bryan-aguilar) [reply .github/workflows/nightly-upstream-snapshot-build.yml:118]: Was this job recently added to the nightly build workflow? I am having a bit of trouble tracing the git blame back for it.

Comment (PaurushGarg) [reply .github/workflows/nightly-upstream-snapshot-build.yml:118]: Yes I added it recently in [this](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/627) PR. and I realized it was a mistake, so removing it with this PR. 
I think this testing should be part of nightly build, but putting it in `nightlybuild` workflows require some updates in the repo to facilitate this.

**Review by rapphil — APPROVED** [review]

**Pull request merged**